apps/src/aichat/views/ChatWorkspace.tsx

```diff
@@ -50,12 +68,30 @@ const ChatWorkspace: React.FunctionComponent<ChatWorkspaceProps> = ({
     }
   }, [messagesString, isWaitingForChatResponse]);
 
-  const dispatch = useAppDispatch();
+  const selectedStudentName = useMemo(() => {
+    if (viewAsUserId) {
+      const selectedStudent = Object.values(students).find(
+        student => student.id === viewAsUserId
+      );
+      if (selectedStudent) {
+        return getShortName(selectedStudent.name);
+      }
+    }
+    return null;
+  }, [viewAsUserId, students]);
 
-  const onCloseWarningModal = useCallback(
-    () => dispatch(setShowWarningModal(false)),
-    [dispatch]
-  );
+  useEffect(() => {
+    // If a teacher is viewing workspace as a student when level first loads (user_id param included in url)
+    // or from when viewing workspace as a participant, default to the student chat history view.
```

Comment:
A little confused by this - when you say "viewing the workspace as a participant" are you referring to the teacher clicking the "Student" toggle under "View as" in the teacher panel? Or the `WORKSPACE_VIEW_MODE.PARTICIPANT` referring to the state of this specific view?

In either case though - couldn't we just determine which workspace view mode to show purely from `viewAsUserId` (i.e. if `viewAsUserId` is defined, show the tabs, if not only show the chat)?

Reply:
// If a teacher is viewing workspace as a student when level first loads (user_id param included in url)
    // or from when viewing workspace as a participant, default to the student chat history view.

I think what I mean was:
- when a teacher was viewing level, clicked on the Teacher Panel, and then clicked on 'Me'
- or when a teacher was on a level with just the level path.

Either way, I agree it's easier to just use `viewAsUserId` to show tabs or not. Just pushed the refactor. Thanks!